src/librustc/mir/mod.rs

```diff
@@ -223,10 +223,10 @@ impl<'tcx> Mir<'tcx> {
         } else if self.local_decls[local].name.is_some() {
             LocalKind::Var
         } else {
-            debug_assert!(
```

Comment:
I'd say just delete the `debug_assert!`, there's no value in leaving it in commented out